cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java

```diff
@@ -111,37 +112,88 @@ private static AmazonDynamoDB defaultDynamoDbClient() {
-        return publication.getAdditionalIdentifiers()
-                   .stream()
-                   .filter(CristinEntryEventConsumer::isCristinAdditionalIdentifier)
-                   .findFirst()
-                   .map(AdditionalIdentifier::getValue);
+    private Publication persistConversionReports(PublicationRepresentations publicationRepresentations) {
+        persistCristinIdentifierInFileNamedWithPublicationIdentifier(publicationRepresentations);
+        persistPartOfCristinIdentifierIfPartOfExists(publicationRepresentations);
+        return publicationRepresentations.getPublication();
     }
 
-    private static boolean isCristinAdditionalIdentifier(AdditionalIdentifier additionalIdentifier) {
-        return CristinObject.IDENTIFIER_ORIGIN.equals(additionalIdentifier.getSource());
+    private PublicationRepresentations mapEventToNvaPublication(
+        FileContentsEvent<JsonNode> eventbody) {
+        var cristinObject = parseCristinObject(eventbody);
+        var publication = cristinObject.toPublication();
+        return new PublicationRepresentations(cristinObject, publication, eventbody);
     }
 
-    private Publication persistCristinIdentifierInFileNamedWithPublicationIdentifier(Publication publication,
-                                                                                     FileContentsEvent<JsonNode> eventBody) {
-        var cristinIdentifier =
-            getCristinIdentifier(publication);
-        var fileUri = constructSuccessFileUri(eventBody, publication);
+    private void persistPartOfCristinIdentifierIfPartOfExists(
+        PublicationRepresentations publicationRepresentations) {
+        if (cristinObjectIsPartOfAnotherPublication(publicationRepresentations.getCristinObject())) {
+            persistPartOfCristinIdentifierWithPublicationId(publicationRepresentations);
+        }
+    }
+
+    private boolean cristinObjectIsPartOfAnotherPublication(CristinObject cristinObject) {
+        return nonNull(cristinObject.getBookOrReportPartMetadata()) && nonNull(
+            cristinObject.getBookOrReportPartMetadata().getPartOf());
+    }
+
+    private String getPartOfCristinIdentifier(CristinObject cristinObject) {
+        return cristinObject.getBookOrReportPartMetadata().getPartOf();
+    }
+
+    private void persistPartOfCristinIdentifierWithPublicationId(
+        PublicationRepresentations publicationRepresentations) {
+        var partOf = createPartOf(publicationRepresentations);
+        var fileUri = constructPartOfFileUri(publicationRepresentations);
+        var s3Driver = new S3Driver(s3Client, fileUri.getUri().getHost());
+        attempt(() -> s3Driver.insertFile(fileUri.toS3bucketPath(), partOf.toJsonString())).orElseThrow();
+    }
+
+    private NvaPublicationPartOfCristinPublication createPartOf(PublicationRepresentations publicationRepresentations) {
+        var publicationIdentifier = publicationRepresentations.getPublication().getIdentifier().toString();
```

Comment:
I believe in general, the internals of the access to values in the contents of the PublicationRepresentations should be hidden — for the methods here, they should not need to know the internal structures as it makes the code more intertwined than need be.

Reply:
fikset